liblangutil/Scanner.cpp

```diff
@@ -967,7 +967,17 @@ tuple<Token, unsigned, unsigned> Scanner::scanIdentifierOrKeyword()
 	while (isIdentifierPart(m_char) || (m_char == '.' && m_kind == ScannerKind::Yul))
 		addLiteralCharAndAdvance();
 	literal.complete();
-	return TokenTraits::fromIdentifierOrKeyword(m_tokens[NextNext].literal);
+	auto const token = TokenTraits::fromIdentifierOrKeyword(m_tokens[NextNext].literal);
+	if (m_kind == ScannerKind::Yul)
+	{
+		// Turn Solidity identifier into a Yul keyword
+		if (m_tokens[NextNext].literal == "leave")
+			return std::make_tuple(Token::Leave, 0, 0);
+		// Turn non-Yul keywords into identifiers.
+		if (!TokenTraits::isYulToken(std::get<0>(token)))
```

Comment:
What about solidity keywords that are non-yul keywords that are separated by `.`? For example `storage.memory`?

Reply:
In Yul mode they are parsed as an identifier already (the "supportsPeriodInIdentifier").